cmake/sdk-support.cmake

```diff
@@ -0,0 +1,128 @@
+function(get_relative_install_path var)
+  file(RELATIVE_PATH install_path ${CMAKE_SOURCE_DIR} ${CMAKE_CURRENT_LIST_DIR})
+  if(NOT install_path)
+    set(install_path .)
+  endif()
+  set(${var} ${install_path} PARENT_SCOPE)
+endfunction()
+
+function(llvm_mos_sdk_add_include_files)
+  get_relative_install_path(inc_path)
+  install(FILES ${ARGN} DESTINATION ${inc_path})
+endfunction()
+
+function(llvm_mos_sdk_add_linker_scripts)
+  get_relative_install_path(script_path)
+  install(FILES ${ARGN} DESTINATION ${script_path})
+endfunction()
+
+function(llvm_mos_sdk_install_targets)
+  get_relative_install_path(lib_path)
+  install(TARGETS ${ARGN} DESTINATION ${lib_path})
+endfunction()
+
+function(llvm_mos_sdk_track_platform_target name)
+  set_property(GLOBAL APPEND PROPERTY LLVM_MOS_CURRENT_PLATFORM_TARGETS ${name})
+endfunction()
+
+function(llvm_mos_sdk_add_crt)
+  set(target_name ${LLVM_MOS_CURRENT_PLATFORM}_crt)
+  add_library(${target_name} STATIC $<TARGET_PROPERTY:common_crt,INTERFACE_SOURCES> ${ARGN})
+  set_target_properties(${target_name} PROPERTIES OUTPUT_NAME crt)
+  llvm_mos_sdk_install_targets(${target_name})
+  llvm_mos_sdk_track_platform_target(${target_name})
+endfunction()
+
+function(llvm_mos_sdk_add_c)
+  set(target_name ${LLVM_MOS_CURRENT_PLATFORM}_c)
+  add_library(${target_name} STATIC $<TARGET_PROPERTY:common_c,INTERFACE_SOURCES> ${ARGN})
+  set_target_properties(${target_name} PROPERTIES OUTPUT_NAME c)
+  llvm_mos_sdk_install_targets(${target_name})
+  llvm_mos_sdk_track_platform_target(${target_name})
+endfunction()
+
+function(llvm_mos_sdk_add_object_file name src)
+  set(target_name ${LLVM_MOS_CURRENT_PLATFORM}_${name})
+
+  if (ARGC GREATER 2)
+    set(outname ${ARGV2})
+  else()
+    set(outname ${target_name})
+  endif()
+
+  # Even though we just need the object file, build a full static library
+  # so POST_BUILD hook works (just ignore the .a, it won't be installed).
+  add_library(${target_name} STATIC ${src})
+  get_relative_install_path(lib_path)
+  install(FILES $<TARGET_OBJECTS:${target_name}> DESTINATION ${lib_path} RENAME ${outname})
+  llvm_mos_sdk_track_platform_target(${target_name})
+
+  # Create symlink (or copy) to the object file on each build.
+  add_custom_command(TARGET ${target_name} POST_BUILD COMMAND ${CMAKE_COMMAND} ARGS -DNAME=${outname}
```

Comment:
Can you add a bit more information to the comment describing the purpose of this symlink. It doesn't seem to immediately suggest itself from the surrounding context.

Reply:
Done